src/components/Button.tsx

```diff
@@ -65,35 +74,74 @@ const gradientDuoToneClasses: Record<GradientDuoTone, string> = {
     'text-gray-900 bg-gradient-to-r from-red-200 via-red-300 to-yellow-200 hover:bg-gradient-to-bl focus:ring-4 focus:ring-red-100 dark:focus:ring-red-400',
 };
 
+const sizeClasses: Record<Size, string> = {
+  extraSmall: 'text-xs px-3 py-2',
+  small: 'text-sm px-3 py-2',
+  medium: 'text-sm px-5 py-2.5',
+  large: 'text-base px-5 py-3',
+  extraLarge: 'text-base px-6 py-3.5',
+};
+
 export const Button: FC<ButtonProps> = ({
   children,
   pill,
   outline,
+  disabled = false,
+  loader = false,
+  iconButton = false,
+  label,
+  size = 'medium',
+  icon: Icon,
+  iconPosition,
   color = 'blue',
   gradientMonochrome,
   gradientDuoTone,
   ...props
 }) => {
   return (
     <button
+      disabled={loader || disabled}
       className={classNames(
-        'flex items-center justify-center p-0.5 text-sm text-center font-medium group',
+        'flex h-min items-center justify-center p-0.5 text-center font-medium group',
         pill ? 'rounded-full' : 'rounded-lg',
         !gradientMonochrome && !gradientDuoTone && colorClasses[color],
         !gradientDuoTone && gradientMonochrome && gradientMonochromeClasses[gradientMonochrome],
         gradientDuoTone && gradientDuoToneClasses[gradientDuoTone],
+        {
+          'bg-blue-400 dark:bg-blue-500 hover:bg-blue-400 dark:hover:bg-blue-400 cursor-not-allowed': disabled,
+        },
       )}
       type="button"
       {...props}
     >
       <span
-        className={classNames('px-5 py-2.5 rounded-md', {
+        className={classNames('flex', size && sizeClasses[size], {
```

Comment:
Looking at the signature of the component the `size` property is always defined since it has the default value `medium`, so we can get rid of the `size &&` part